flang/include/flang/Tools/CLOptions.inc

```diff
@@ -254,6 +254,24 @@ inline void createHLFIRToFIRPassPipeline(
   pm.addPass(hlfir::createConvertHLFIRtoFIRPass());
 }
 
+/// Create a pass pipeline for handling certain OpenMP transformations needed
+/// prior to FIR lowering.
+///
+/// WARNING: These passes must be run immediately after the lowering to ensure
+/// that the FIR is correct with respect to OpenMP operations/attributes.
```

Comment:
Can you add the WARNING also to the places where they are called so that it is more visible and others might not add a pass before it?

Reply:
Done, thank you very much for the quick review!